File: c++/method_fail_2.cpp
// @expect error

#include <smack.h>//<cassert>

class Car {
  public:
    int numWheels() {
        return 4;
    }
    int numHeadlights() {
        return 2;
    }
    int milageLeft(int miles) {
        return 300000 - miles;
    }
};

int main(void) {
    Car *c = new Car();
    assert(c->numWheels() == 4);
    assert(c->numHeadlights() != 2);
    assert(c->milageLeft(250000) == 50000);
    delete c;
    return 0;
}

File: swift/method_fail_2.swift
// @expect error

class Car {
    func numWheels() -> Int {
        return 4
    }

    func numHeadlights() -> Int {
        return 2
    }

//    func milageLeft(miles: Int) -> Int {
//        return 300_000 - miles
//    }
}

let car = Car()
__VERIFIER_assert(car.numWheels() == 4)
//__VERIFIER_assert(car.numHeadlights() != 2)
//__VERIFIER_assert(car.milageLeft(miles: 250_000) != 50_000)
